contracts/4/BridgedTokenOWL.sol

```diff
@@ -0,0 +1,39 @@
+pragma solidity ^0.4.24;
+
+import "../../vendor/tokenbridge-contracts/contracts/PermittableToken.sol";
+
+/// @dev A contract to implement OWL tokens on a bridged chain (e.g. XDai).
+/// Implements "burning" by allocating the amount to a dedicated receiver,
+/// which can transfer it to the native chain (e.g. Mainnet) and complete
+/// the burn there.
+contract BridgedTokenOwl is PermittableToken {
+    address public burnReceiver;
+
+    event Burnt(address indexed from, address indexed user, uint256 amount);
+
+    constructor(
+        string memory _name,
+        string memory _symbol,
+        uint8 _decimals,
+        uint256 _chainId,
+        address _burnReceiver
+    ) public PermittableToken(_name, _symbol, _decimals, _chainId) {
+        burnReceiver = _burnReceiver;
+    }
+
+    /// @dev Burns OWL.
+    /// @param user Address of OWL owner
+    /// @param amount Amount of OWL to be burnt
+    function burnOWL(address user, uint256 amount) public {
```

Comment:
Should we also overwrite the `burn` method (https://github.com/OpenZeppelin/openzeppelin-contracts/blob/v1.12.0/contracts/token/ERC20/BurnableToken.sol#L18)? Or is it intended that you can still normally burn your owl on the bridged chain?

Reply:
The other `burn` method is needed to bring the tokens back on the mainnet chain.

Reply:
but shouldn't be the only one to be able to burn owl tokens on the bridged chain be the burnReceiver?

Reply:
No, if you have bridged 1k OWL into xDAI and want to take a part of it back to mainnet you need to call the burn function as well. The burnReceiver just brings them on mainnet to another contract under their control which then can call the real `burnOwl` function on mainnet.

Reply:
Ahh I misunderstood the purpose of the function ... so this is so that it is also possible to burn owl also on xdai (which is required for GP). This is required as the normal `burn` is used to transfer the funds to mainnet again, right?